lib/src/_shared/hooks/usePickerState.ts

```diff
@@ -24,112 +24,82 @@ const valueToDate = (
   const utils = useUtils();
   const date = valueToDate(utils, props);
-  const acceptedDateRef = useRef(date);
   const format = props.format || options.getDefaultFormat();
 
-  return { acceptedDateRef, date, format };
+  return { date, format };
 }
 
-function makeControlledOpenProps(props: BasePickerProps) {
-  return {
-    isOpen: props.open!,
-    setIsOpen: (newIsOpen: boolean) => {
-      return newIsOpen ? props.onOpen && props.onOpen() : props.onClose && props.onClose();
-    },
-  };
-}
-
-/* eslint-disable react-hooks/rules-of-hooks */
-function useOpenState(props: BasePickerProps) {
-  if (props.open !== undefined && props.open !== null) {
-    return makeControlledOpenProps(props);
-  }
-
-  const [isOpen, setIsOpenState] = useState(false);
-  // prettier-ignore
-  const setIsOpen = useCallback((newIsOpen: boolean) => {
-    setIsOpenState(newIsOpen);
-
-    return newIsOpen
-      ? props.onOpen && props.onOpen()
-      : props.onClose && props.onClose()
-  }, [props]);
-
-  return { isOpen, setIsOpen };
-}
-/* eslint-enable react-hooks/rules-of-hooks */
-
 export function usePickerState(props: BasePickerProps, options: StateHookOptions) {
   const utils = useUtils();
   const { isOpen, setIsOpen } = useOpenState(props);
-  const { acceptedDateRef, date, format } = useDateValues(props, options);
+  const { date, format } = useDateValues(props, options);
+  const [pickerDate, setPickerDate] = useState(date);
 
-  if (!isOpen) {
-    // if value was changed in closed state treat it as accepted
-    acceptedDateRef.current = date;
-  }
-
-  const validationError = validate(props.value, utils, props);
   useEffect(() => {
-    if (validationError && props.onError) {
-      props.onError(validationError, props.value);
+    // if value was changed in closed state - treat it as accepted
+    if (!isOpen && !utils.isEqual(pickerDate, date)) {
+      setPickerDate(date);
     }
-  }, [props, validationError]);
-
-  const inputProps = useMemo(
-    () => ({
-      validationError,
-      onClick: () => !props.disabled && setIsOpen(true),
-      inputValue: getDisplayDate(
-        acceptedDateRef.current,
-        format,
-        utils,
-        props.value === null,
-        props
-      ),
-    }),
-    [acceptedDateRef, format, props, setIsOpen, utils, validationError]
-  );
+  }, [date, isOpen, pickerDate, utils]);
 
-  // prettier-ignore
-  const acceptDate = useCallback((acceptedDate: MaterialUiPickersDate) => {
-    acceptedDateRef.current = acceptedDate;
+  const acceptDate = useCallback(
+    (acceptedDate: MaterialUiPickersDate) => {
+      setIsOpen(false);
+      props.onChange(acceptedDate);
 
-    setIsOpen(false);
-    props.onChange(acceptedDate);
-
-    if (props.onAccept) {
-      props.onAccept(acceptedDate);
-    }
-  }, [acceptedDateRef, setIsOpen, props]);
+      if (props.onAccept) {
+        props.onAccept(acceptedDate);
+      }
+    },
+    [setIsOpen, props]
```

Comment:
AFAIU, the `props` object always changes. It's possible, in theory, that the component that passes the props has them memoized, but in all normal circumstances (using JSX, or call sites like [this](https://github.com/mui-org/material-ui-pickers/blob/68779c7237976c8f56278322885aee980cbc3ed5/lib/src/_shared/hooks/useKeyboardPickerState.ts#L51)), the `props` object will be a new object. And `useCallback` [only memoizes](https://github.com/facebook/react/blob/425473f43fd95a564f07ffcf6dbd105cef9f78b9/packages/react-reconciler/src/ReactFiberHooks.js#L342) when the `deps` are the same according to `Object.is()`, so the `useCallback()` here is ineffective. It can either be removed (if we don't care about memoization) or be fixed:

```
const {onAccept, onChange} = props;
const acceptDate = useCallback(
  (acceptedDate: MaterialUiPickersDate) => {
    setIsOpen(false);
    onChange(acceptedDate);
     if (onAccept) {
      onAccept(acceptedDate);
    }
  },
  [setIsOpen, onAccept, onChange]
);
```

If you agree (I might be wrong, I'm very new to hooks), I can send a PR for this and similar occurrences.

Reply:
Yep, that will be more efficient

Reply:
It will not be changed if only internal state changed.  It anyway it will be cool to have more clear deps for top level hook